### PR TITLE
uncordon on upgrade job completion

### LINKED ISSUE #9: uncordon after successful upgrade
**Is your feature request related to a problem? Please describe.**
After a job has completed successfully it is left to a human operator to uncordon an upgraded node (if it was cordoned or drained).

**Describe the solution you'd like**
After a plan has been successfully applied to a node with `cordon`/`drain` configured the controller should uncordon the node.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by dweomer**
Sets `node.spec.unschedulable` to `false` after a job completes successfully.
This happens in the same operation that sets the **Plan**-specific tag on job success.

Fixes #9

### CODE REVIEW

File: pkg/upgrade/plan/plan.go [hunk 1]
```diff
@@ -132,6 +131,9 @@ func RegisterHandlers(ctx context.Context, serviceAccountName, controllerNamespa
 								return obj, err
 							}
 							node.Labels[planLabel] = version
+							if node.Spec.Unschedulable {
```
Comment (dweomer) [inline pkg/upgrade/plan/plan.go]: Should we mark the node as schedulable no matter what (as we are doing here) or should we only flip it back if cordschedulableon is true or drain is not nil?

File: pkg/upgrade/plan/plan.go [hunk 2]
```diff
@@ -125,13 +124,20 @@ func RegisterHandlers(ctx context.Context, serviceAccountName, controllerNamespa
 				defer plans.Enqueue(obj.Namespace, planName)
 				if obj.Status.Succeeded == 1 {
 					planLabel := upgradeapi.LabelPlanName(planName)
-					if version, ok := obj.Labels[planLabel]; ok {
+					if planHash, ok := obj.Labels[planLabel]; ok {
 						if nodeName, ok := obj.Labels[upgradeapi.LabelNode]; ok {
 							node, err := nodes.Get(nodeName, metav1.GetOptions{})
```
Comment (galal-hussein) [inline pkg/upgrade/plan/plan.go]: just a nit: shouldn't we use a node cache to be quicker on a large scale here?

Comment (dweomer) [reply pkg/upgrade/plan/plan.go]: @galal-hussein plan cache too then I would think.

Comment (dweomer) [reply pkg/upgrade/plan/plan.go]: pushed new commit that uses caches

**Pull request merged**